### PR TITLE
Pull Request dev_1.1

### PR DESCRIPTION
**Pull request by erikjohnmoreno**

### CODE REVIEW

File: app/controllers/comment_controller.php [hunk 1]
```diff
@@ -10,14 +10,18 @@ public function view()
     {  
         $session_id = $_SESSION['id'];
         $thread = Thread::get(Param::get('thread_id'));
```
Comment (romeoartista) [inline app/controllers/comment_controller.php]: Define a default value for `$thread` first to allow validation before passing it to the `Thread::get()` method.

``` php
$thread_id = Param::get('thread_id');

if (!$thread_id) {
    // Error handling
}

$thread = Thread::get($thread_id);
```

File: app/controllers/comment_controller.php [hunk 2]
```diff
@@ -10,14 +10,18 @@ public function view()
...
+        $comments = $comment->getCommentsByThread($thread->id);//getting all comments on the selected thread
+        $users = $comment->getByUser();//getting all user
+        $_SESSION['last_page'] = count(array_chunk($comments, self::MAX_COMMENT_PER_PAGE));//getting the last page
         $current = max(Param::get('page'), SimplePagination::MIN_PAGE_NUM);
         $pagination = new SimplePagination($current, self::MAX_COMMENT_PER_PAGE);
         $remaining_comments = array_slice($comments, $pagination->start_index + SimplePagination::MIN_PAGE_NUM);
         $pagination->checkLastPage($remaining_comments);
         $page_links = createPaginationLinks(count($comments), $current, $pagination->count,'thread_id='.$thread->id);
         $comments = array_slice($comments, $pagination->start_index, $pagination->count);
+        $_SESSION['thread_id'] = $thread->id;
```
Comment (romeoartista) [inline app/controllers/comment_controller.php:27]: I think it is better to use the `$_GET` variable than `$_SESSION` in passing values to and from pages.

**Comment by romeoartista** [comment]
In `app/controllers/comment_controller.php` line 31, can be changed to something like in [this comment](https://github.com/erikjohnmoreno/board_exercises/pull/2/files#r24557679).

File: app/controllers/comment_controller.php (hunk 2 above)
Comment (erikjohnmoreno) [inline app/controllers/comment_controller.php:27]: I'm using the $_SESSION po for redirection :bow:

File: app/controllers/comment_controller.php (hunk 1 above)
Comment (erikjohnmoreno) [inline app/controllers/comment_controller.php]: done po :bow:

File: app/models/comment.php [hunk 3]
```diff
@@ -111,8 +145,118 @@ public function delete($comment_id)
...
+        try {
+            $db = DB::conn();
+            $db->begin();
+            if ($this->isliked($this->comment_id, $user_id)) {
+                $db->query('DELETE FROM liked 
+                            WHERE comment_id = ? 
+                            AND user_id = ?', array($this->comment_id, $user_id));
+
+                $db->query('UPDATE comment 
+                            SET likes = likes - 1
```
Comment (charmgranada) [inline app/models/comment.php:212]: same comment as: https://github.com/dandiesta/board_exercise/pull/2#discussion_r24567499

File: app/models/comment.php [hunk 4]
```diff
@@ -20,23 +20,24 @@ class Comment extends AppModel
...
     {
-        $comments = array();
+        return user::getAllUser();
+    } 
 
+    /**
+    * get comments in a thread
+    * @param $thread_id
+    */
+    public function getCommentsByThread($thread_id)
```
Comment (charmgranada) [inline app/models/comment.php]: Please rename all functions in this class :bow: 
Remove redundancy by not using 'Comment' too much as it is already the name of the class :bow:

Comment (erikjohnmoreno) [inline app/models/comment.php]: done po :bow:

**Pull request merged**